tests/_internal/test_resolve_target.py

```diff
@@ -0,0 +1,91 @@
+import pathlib
+from mbed_tools_lib.exceptions import ToolsError
+from pyfakefs.fake_filesystem_unittest import Patcher
+from unittest import TestCase, mock
+
+from tests.factories import CandidateDeviceFactory
+from mbed_devices._internal.htm_file import HTMFileContentsParser, OnlineId
+from mbed_devices._internal.resolve_target import (
+    _determine_target_resolver,
+    _get_all_htm_files_content_parsers,
+    resolve_target,
+)
+
+
+@mock.patch("mbed_devices._internal.resolve_target._determine_target_resolver")
+class TestResolveTarget(TestCase):
+    def test_uses_target_resolver_to_determine_targets(self, _determine_target_resolver):
+        resolver = mock.Mock()
+        _determine_target_resolver.return_value = resolver
+        candidate = CandidateDeviceFactory()
+
+        self.assertEqual(resolve_target(candidate), resolver.return_value)
+        _determine_target_resolver.assert_called_once_with(candidate.mount_points)
+
+    def test_handles_tools_errors_raised_by_resolver(self, _determine_target_resolver):
+        resolver = mock.Mock(side_effect=ToolsError)
+        _determine_target_resolver.return_value = resolver
+        candidate = CandidateDeviceFactory()
+
+        self.assertIsNone(resolve_target(candidate))
+
+
+@mock.patch("mbed_devices._internal.resolve_target._get_all_htm_files_content_parsers")
+class TestDetermineTargetResolver(TestCase):
+    @mock.patch("mbed_devices._internal.resolve_target.get_target_by_product_code")
```

Comment:
Would it be worth spec-setting this mock? Then we would be alerted if there were changes in the mbed-targets repo.

Reply:
Sadly, I don't think you can `spec_set` functions (checking kwargs is not supported). I'd like to be wrong. :) 

If you're talking about spec_setting the `mbed_targets` module import, it will explode if `get_target_by_*` is not exported by `mbed_targets` w/o spec_setting.

Reply:
Ah, yeah I was hoping for args checking but you are right that doesn't seem to be a thing. All good.